system/xmllib/libxml_xpathprocessor.cpp

```diff
@@ -1624,44 +1612,41 @@ static const char *queryAttrValue(xmlNodePtr node, const char *name)
     return (const char *) att->children->content;
 }
 
-class CEsdlScriptContext : public CInterfaceOf<IEsdlScriptContext>
+class CXmlDocContext : public CInterfaceOf<ISectionalXmlDocModel>
 {
 private:
-    void *espCtx = nullptr;
+    void *usageContext = nullptr;
```

Comment:
We always know the lifetime of the usageContext will be greater than the CXmlDocContext because it owns the DocContext- at least in the EsdlScriptContext implementation?

Reply:
I suppose we don't know it any more (or less) than we did before. The caller has always been required to ensure the object passed had a longer life than the context/model holding the pointer. There has never been a technical restriction preventing a script context from being retained across transactions, so even holding an ESP context was a theoretical risk.

Is the risk serious enough to consider making this a shared pointer? Such a change would give us two objects, each sharing the other, that would require extra work to ensure they are released.

Reply:
Good inadvertent catch of the class naming issue - I forgot to rename the implementation of doc model interface.

Reply:
After offline discussion with @afishbeck, the document model is to have no knowledge of what the value represents, including whether it is even a pointer. `usageContext` is renamed `userData` and a comment is added making value "ownership" the model owner's responsibility.